jetty-websocket/websocket-javax-common/src/main/java/org/eclipse/jetty/websocket/javax/common/JavaxWebSocketSession.java

```diff
@@ -194,9 +192,18 @@ public void close() throws IOException
     @Override
     public void close(CloseReason closeReason) throws IOException
     {
-        try (SharedBlockingCallback.Blocker blocker = blocking.acquire())
+        if (!closeInitiated.compareAndSet(false, true))
+            return;
+
+        try
+        {
+            BlockingCallback b = new BlockingCallback();
+            coreSession.close(closeReason.getCloseCode().getCode(), closeReason.getReasonPhrase(), b);
+            b.block();
+        }
+        catch (RuntimeException e)
         {
-            coreSession.close(closeReason.getCloseCode().getCode(), closeReason.getReasonPhrase(), blocker);
+            LOG.warn(e);
```

Comment:
I don't think we want to log and swallow the exception here. I would remove the `catch` - it's a `RuntimeException` and if it happens let it flow out.

Reply:
If the `CoreSession` was already closed (we didn't initiate the close here so `closeInitiated` is still false), then closing would throw an ISE because we have already been closed.

Reply:
But this is more of an issue on the jetty API case where the close signature does not throw any exceptions.

Reply:
@lachlan-roberts concurrent closes should not throw IMHO, nto even from core. They are possible and they are not rare at all, so I'd rather fix core to not throw ISE on concurrent closes, but just make it idempotent (i.e. no operation if already closed).

Reply:
@sbordet the `CoreSession` close takes a callback, we fail to callback which indicates that the frame was not sent. If you don't care about whether the frame was sent or not you can give it `NOOP` and ignore the result.

In that case for this method we could not use the `BlockingCallback` altogether and just go:
```java
coreSession.close(closeCode, reasonPhrase, NOOP);
```